velox/dwio/parquet/tests/reader/ParquetQueryTest.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/init/Init.h>
+#include <vector>
+
+#include "velox/common/base/Fs.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/file/FileSystems.h"
+#include "velox/dwio/common/tests/utils/DataFiles.h"
+#include "velox/dwio/parquet/RegisterParquetReader.h"
+#include "velox/dwio/parquet/duckdb_reader/ParquetReader.h"
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/exec/tests/utils/TpchQueryBuilder.h"
+#include "velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/parse/TypeResolver.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::parquet;
+
+static const int kNumDrivers = 4;
+
+class ParquetQueryTest : public testing::Test {
+ protected:
+  // Setup a DuckDB instance for the entire suite and load TPC-H data with scale
+  // factor 0.01.
+  static void SetUpTestSuite() {
+    if (duckDb_ == nullptr) {
+      duckDb_ = std::make_shared<DuckDbQueryRunner>();
+    }
+    functions::prestosql::registerAllScalarFunctions();
+    parse::registerTypeResolver();
+    filesystems::registerLocalFileSystem();
+    registerParquetReaderFactory(parquet::ParquetReaderType::NATIVE);
+
+    auto hiveConnector =
+        connector::getConnectorFactory(
+            connector::hive::HiveConnectorFactory::kHiveConnectorName)
+            ->newConnector(kHiveConnectorId, nullptr);
+    connector::registerConnector(hiveConnector);
+    tempDirectory_ = exec::test::TempDirectoryPath::create();
+  }
+
+  std::string getExampleFilePath(const std::string& fileName) {
+    return facebook::velox::test::getDataFilePath(
+        "velox/dwio/parquet/tests/reader", "../examples/" + fileName);
+  }
+
+  std::shared_ptr<Task> assertQuery(
```

Comment:
Looks like a lot of code in this test is a copy-paste from velox/dwio/parquet/tests/ParquetTpchTestBase.h

It would be nice to refactor to avoid copy-paste.

CC: @majetideepak